encryption.go

```diff
@@ -13,6 +13,7 @@ import (
 // that produces an encryption key via X25519.
 type X25519Producer interface {
 	X25519EncryptionKey() ([]byte, error)
+	PreviousKey() (string, []byte, error)
```

Comment:
I think this should be PreviousX25519EncryptionKey. The reason it's called X25519EncryptionKey instead of just EncryptionKey is that if we ever support more than one type and can accept a variety of interfaces to get encryption keys, we could potentially clash if each one had a method called `EncryptionKey`. By putting X25519 in there we ensure that if we e.g. have a higher level interface embedding this and another that they won't clash.

It can be worked around, but feels cleaner to just be explicit.